central/src/main/webapp/components/ModelNavigator.jsx

```diff
@@ -86,25 +98,117 @@ export default function ModelNavigator(props) {
+                                    <TextField
+                                        id="name-search"
+                                        label="Enter Name"
+                                        value={nameValue}
+                                        onChange={modelFilterOnChange}
+                                        inputProps={{style: { backgroundColor:"white"},}}
+                                    />
+                                </TreeItem>
+                                <TreeItem nodeId="Version" label="Version">
+                                    <div onChange={modelVersionFilterOnChange}>
+                                        {modelZooData.map((application) => (
+                                            application.models.map((model) => (
+                                                    handleAdd(model.version)
+                                            ))
+                                        ))}
+
+                                        {versionList.map((version) => (
+                                            <button disabled={versionValue==version} value={version} onClick={modelVersionFilterOnChange} > {version} </button>
+                                        ))}
+                                        <button value="" onClick={modelVersionFilterOnChange} > Clear </button>
+                                    </div>
+                                </TreeItem>
+                                <TreeItem nodeId="Application" label="Application">
+                                    <div onChange={modelApplicationFilterOnChange}>
+                                        {modelZooData.map((application) => (
+                                            <button
+                                                disabled={applicationValue == application.title}
+                                                value={application.title}
+                                                onClick={modelApplicationFilterOnChange}
+                                            >
+
+                                                {application.title}
+
+                                            </button>
+                                        ))}
+                                       <button value='' onClick={modelApplicationFilterOnChange} > Clear </button>
+                                    </div>
+                                </TreeItem>
+                            </TreeView>
+                            <button onClick={modelFilterOnChange} > Search </button>
+                        </TreeItem>
+                    </TreeView>
 
-			<div className={classes.view_root}>
-				<div className={classes.navigator_root}>
 				<TreeView
-					
+
 					defaultCollapseIcon={<ExpandMoreIcon />}
 					defaultExpandIcon={<ChevronRightIcon />}
 				>
-					{modelZooData.map((application) => (
-						<TreeItem nodeId={application.key} label={application.title}>
```

Comment:
Can you keep the application trees when displaying the models? When having no search filters, which is probably the most common way users are going to use this page, it helps a lot in organizing the full list of models.

Reply:
I have now made it to where if there is no search filters are being used it will display the model list with the application trees and if it is filtered it will begin to show each model. Hopefully that addresses this comment.